.buildkite/hooks/pre-command

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -euo pipefail
+
+source .buildkite/scripts/common.sh
+
+DOCKER_REGISTRY_SECRET_PATH="kv/ci-shared/platform-ingest/docker_registry_prod"
+DOCKER_REGISTRY="docker.elastic.co"
+export DOCKER_REGISTRY="${DOCKER_REGISTRY}"
+
+if [[ "$BUILDKITE_STEP_KEY" == "publish" ]]; then
```

Comment:
nit. Add also the "$BUILDKITE_PIPELINE_SLUG" == "fleet-server" 

```suggestion
if [[ "$BUILDKITE_PIPELINE_SLUG" == "fleet-server" && "$BUILDKITE_STEP_KEY" == "publish" ]]; then
```

This pre-command is going to be used by all the pipelines defined in the repository. It is better to narrow here where these secrets are going to be retrieved and used.

Reply:
Fixed.
I thought if we have only one pipeline we shouldn't define extra conditions to check the pipeline because it could confuse smb with the questions "Why?" and "Where is the second pipeline?"
And probably it's better to add a comment with an example of using the multi-pipeline solution.
What do you think, @mrodm ?

Reply:
At least, there is one more CI job to migrate to Buildkite from fleet-server at some point:
https://github.com/elastic/fleet-server/blob/main/.ci/packaging.groovy

Not checked, but it is likely that we need to add more code in these hooks for some steps.

We could add some comment about the reasoning of this if that could help other users, sure. The main reason for me is to narrow the scope of the secrets to the steps where they are need. And to identify them it would be needed both pipeline slug and step key.